### PR TITLE
HPCC-29370 Revise validateDropZonePath() and related code

### PR DESCRIPTION
**Pull request by wangkx**
1. Add findDropZonePlaneWithName();
2. Rename the validateDropZonePath to validateDropZoneHostAndPath.
3. Call the findDropZonePlaneWithName in validateDropZoneHostAndPath.
4. Revise validateDropZonePath related code in downloadFile(), onStartUpload(), and onFileList().

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/smc/SMCLib/TpCommon.cpp [hunk 1]
```diff
@@ -119,41 +119,21 @@ extern TPWRAPPER_API bool matchNetAddressRequest(const char* netAddressReg, bool
...
-        ForEachItemIn(ii, tpMachines)
-        {
-            if (matchNetAddressRequest(netAddr, isIPAddressReq, tpMachines.item(ii)))
-                return true;
-        }
-    }
-    return false;
+    StringBuffer path(pathToCheck);
+    addPathSepChar(path);
+    Owned<IPropertyTree> plane = findDropZonePlaneWithName(dropZoneName, path, hostToCheck, isIPAddress(hostToCheck), false);
```
Comment (jakesmith) [inline esp/smc/SMCLib/TpCommon.cpp:135]: not keen on adding this (backward compatibility) functionality at a lower level.
I think this code should be:
```
    if (isEmptyString(dropZoneName))
    {
        Owned<IPropertyTree> plane = findDropZonePlane(path, hostToCheck, isIPAddress(hostToCheck), false);
        return nullptr != plane;
    }
    else
    {
        Owned<IPropertyTree> plane = getDropZonePlane(dropZoneName);
        if (nullptr == plane)
            return false;
        return validateDropZone(plane, path, hostToCheck, isIPAddress(hostToCheck));
    }
```

With validateDropZone being a new function in dautils.hpp which is code extracted out of the existing findDropZonePlane, so it calls it, and here calls it.

**Review by jakesmith — CHANGES_REQUESTED** [review]
@wangkx - looks a lot cleaner, but see comment re. splitting findDropZonePlane / adding a validateDropZone

**Comment by jakesmith** [comment]
@wangkx - also, should retarget this against 9.2/master

**Comment by wangkx** [comment]
@jakesmith Please review.

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good, please squash.

**Comment by wangkx** [comment]
@ghalliday please merge this PR.

File: esp/services/ws_fs/ws_fsBinding.cpp [hunk 2]
```diff
@@ -399,52 +399,25 @@ int CFileSpraySoapBindingEx::downloadFile(IEspContext &context, CHttpRequest* re
...
-        if (nameStr.length() < 1)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT,"File name not specified.");
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file name %s", nameStr.str());
 
+        if (pathStr.isEmpty())
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Path not specified.");
         char pathSep = '/';
-        if ((osStr.length() > 1) && (atoi(osStr.str())== OS_WINDOWS))
-        {
+        if (osStr.isEmpty() && (atoi(osStr.str())== OS_WINDOWS))
```
Comment (afishbeck) [inline esp/services/ws_fs/ws_fsBinding.cpp:410]: I think you mean "!osStr.isEmpty()"?  For the atoi the string shouldn't be empty.

**Review by afishbeck — COMMENTED** [review]
@wangkx one comment when checking for OS_WINDOWS.

**Review by afishbeck — APPROVED** [review]

**Comment by wangkx** [comment]
@ghalliday please merge this PR.

**Pull request merged**